src/tests/golden/quickfix_test.go

```diff
@@ -0,0 +1,123 @@
+package golden
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/VKCOM/noverify/src/linter"
+	"github.com/VKCOM/noverify/src/linttest"
+)
+
+func TestQuickFix(t *testing.T) {
+	s := newQuickFixTest(t, `testdata/quickfix`)
+	s.runQuickFixTest()
+}
+
+const (
+	expectedExtension = ".fix.expected"
+	fixExtension      = ".fix"
+)
+
+type quickFixTest struct {
+	t      *testing.T
+	folder string
+}
+
+func newQuickFixTest(t *testing.T, folder string) quickFixTest {
+	return quickFixTest{
+		t:      t,
+		folder: folder,
+	}
+}
+
+func openFile(filename string) (f *os.File, found bool, err error) {
+	f, err = os.Open(filename)
```

Comment:
You can use other Open call that creates a file if it doesn't exist. I'm not 100% sure why the hard way is used here.

Reply:
Also, you do open+write+close without extra logic, at least it looks like it.
writeFile can be a function, not openFile. This way your write is more atomic and close can be deferred.
Avoid passing resources that need to be closed when  it's not necessary.

Reply:
We need to know if the `.fix.expected` file exists in order to implement the creation of a ready-made file with the expected code if it does not exist.